File: app/src/main/java/com/example/mobile_challenge/model/ErrorResponse.kt
package com.example.mobile_challenge.model

import kotlinx.serialization.Serializable

@Serializable
class ErrorResponse(
  val success: Boolean,
  val error: ErrorInfo
)

@Serializable
data class ErrorInfo(
  val code: Int,
  val type: String,
  val info: String
)

File: CurrencyConverter/CurrencyConverter/Source/Network/ErrorResponse.swift
//
//  ErrorResponse.swift
//  CurrencyConverter
//
//  Created by Italo Boss on 13/12/20.
//

struct ErrorResponse: Error, Decodable {
    let success: Bool
    let error: ErrorDetail
}

struct ErrorDetail: Decodable {
    let code: Int
    let info: String
}
